RevenueCatUI/UIKit/PaywallViewController.swift

```diff
@@ -150,12 +115,53 @@ public class PaywallViewController: UIViewController {
         super.viewDidDisappear(animated)
     }
 
+    /// - Warning: For internal use only
+    public func update(with offering: Offering) {
+        self.configuration.content = .offering(offering)
+    }
+
+    /// - Warning: For internal use only
+    public func update(with offeringIdentifier: String) {
+        self.configuration.content = .offeringIdentifier(offeringIdentifier)
+    }
```

Comment:
@vegaro this is the new API.

Reply:
Thanks. Couple things I've noticed while testing. This is how it gets called from objc:

```
- (void)setOptions:(NSDictionary *)options{
    NSDictionary *offering = options[@"offering"];
    if (offering && ![offering isKindOfClass:[NSNull class]]) {
        NSString *identifier = offering[@"identifier"];
        if (identifier) {
            [self.footerViewController updateWith:identifier];
        }
    }
}
```

It needs `@objc`. And we might want to update the name at least in objc because right now, if you try to add `@objc` to the one that accepts an `Offering` there's a name colision. There will probably be other parameters added soon.

Reply:
Another thing, when we create the paywall in full screen, we will also need to be able to update the offering. Right now `createPaywallView` in `PaywallProxy` returns a `UIViewController`, we might want to update it to return a `PaywallViewController` instead.

```
    @objc
    public func createPaywallView() -> UIViewController {
        let controller = PaywallViewController()
        controller.delegate = self
        return controller
    }
```

Reply:
If we end up modifying `createPaywallView` we will need https://github.com/RevenueCat/react-native-purchases/pull/860

Reply:
I added the `@objc` annotations 👍🏻  

> Right now createPaywallView in PaywallProxy returns a UIViewController

Yeah let's change that to return `PaywallViewController`.

Reply:
I did that in https://github.com/RevenueCat/purchases-hybrid-common/pull/675